code/modules/food_and_drinks/machinery/microwave.dm

```diff
@@ -89,6 +89,8 @@
 	for(var/obj/item/stock_parts/matter_bin/M in component_parts)
 		max_n_of_items = 10 * M.rating
 		break
+	for(var/thing in component_parts)
+		ingredients -= thing
```

Comment:
we're adding it to ingredients then removing it, instead why should we add it to the list in the first place?
We can either check if the item is edible or part of their circuit's req_components, and to not add it to the ingredient list if so

Reply:
i agree

Reply:
There are non edible items that can be microwaved (ie hide)

 Looking for something a little more elegant.

Reply:
How about a check when inserting something into the microwave if it's a component part?

Reply:
> How about a check when inserting something into the microwave if it's a component part?

If it's.... in the list? Or if it's a part that can be used a component part?

If you mean if it's in the list, that isn't guaranteed at all. At a glace, that'd require changing the order of operations in how machine frames are constructed to start with, which absolutely would need to be tested to make sure it doesn't break some random machine somewhere.

Reply:
So just to double check that we're on the same page with this, you guys want me to switch from using the proc RefreshParts(), which is only called after the parts to a machine changes, with the intent of it updating whatever needs to be updated on the machine, to instead add a for() loop with 4 istype checks to Entered(), meaning it'll be run every single time an item is inserted into a microwave?

All this really added was more robust varedit support lmao. really at that point we might as well just revert the fucking change all together honestly since that's just adding free lag. One is called once a round after science spends the entire round upgrading research, the other is called **every time the chef cooks an item** lol

Reply:
Well my confusion is just that `ingredients` is basically contents, and not the actual list of ingredients